deploy/vagrant/provision/playbook.yaml

```diff
@@ -61,34 +65,79 @@
           --exclude=underFSStorage
       when: (not is_vb) and (not is_git)
 
+    - copy: src=../ufs/{{ ufs }}/compile_tachyon.sh dest=/tmp/compile_tachyon.sh
     - name: compile tachyon
-      script: ../ufs/{{ ufs }}/compile_tachyon.sh
+      command: bash /tmp/compile_tachyon.sh
+      async: 1200
+      poll: 60
+
+    - name: set tachyon/conf/workers tachyon/conf/slaves
+      shell: cp /vagrant/files/workers /tachyon/conf/workers && cp /vagrant/files/workers /tachyon/conf/slaves
+
+    - name: set tachyon-env.sh
+      script: ../ufs/{{ ufs }}/config_tachyon.sh
+    - name: set TACHYON_SSH_FOREGROUND=""
+      shell: sed -i 's/TACHYON_SSH_FOREGROUND="yes"/TACHYON_SSH_FOREGROUND=""/g' /tachyon/conf/tachyon-env.sh
+    - name: set TACHYON_JAR in libexec/tachyon-config.sh
+      shell: tachyon_version=`grep version /tachyon/pom.xml | \
+             head -n1 | tr -d ' ' | \
+             sed 's/<version>//g' | \
+             sed 's/<\/version>//g'` && \
+             sed -i "s/tachyon-.*-jar-with-dependencies.jar/tachyon-${tachyon_version}-jar-with-dependencies.jar/g" /tachyon/libexec/tachyon-config.sh
+
+    - name: clone spark repo
+      git: repo={{ spark_repo }} dest=/spark version={{ spark_version }}
+      when: use_spark
+
+    - name: set current tachyon version in spark/core/pom.xml
+      shell: tachyon_version=`/tachyon/bin/tachyon version | cut -d':' -f2 | tr -d ' '` && \
```

Comment:
Will this work if the user is using a non release branch? I don't think its a good idea to change Spark's Tachyon version in general since the code will be based off the assumed version's APIs.